tst/com/amazon/corretto/crypto/provider/test/RsaGenTest.java

```diff
@@ -221,8 +214,12 @@ private static void assertConsistency(final RSAPublicKey pub, final RSAPrivateCr
         final BigInteger totient = p1.multiply(q1).divide(p1.gcd(q1));
         assertEquals(BigInteger.ONE, e.multiply(d).mod(totient));
 
+        final BigInteger phiGCD = p1.gcd(q1);
+        final BigInteger phi = p1.multiply(q1).divide(phiGCD);
```

Comment:
why divide by `phiGCD` here? `phi(n) = (q-1)(p-1)` per [wikipedia](https://en.wikipedia.org/wiki/RSA_(cryptosystem)#Key_generation):

> In the original RSA paper,[[2]](https://en.wikipedia.org/wiki/RSA_(cryptosystem)#cite_note-rsa-2) the [Euler totient function](https://en.wikipedia.org/wiki/Euler_totient_function) φ(n) = (p − 1)(q − 1) is used instead of λ(n) for calculating the private exponent d.

and elsewhere in that section:

> Compute λ(n), where λ is [Carmichael's totient function](https://en.wikipedia.org/wiki/Carmichael%27s_totient_function). Since n = pq, λ(n) = [lcm](https://en.wikipedia.org/wiki/Least_common_multiple)(λ(p), λ(q)), and since p and q are prime, λ(p) = [φ](https://en.wikipedia.org/wiki/Euler_totient_function)(p) = p − 1, and likewise λ(q) = q − 1. Hence λ(n) = lcm(p − 1, q − 1).

`lcm(x, y) = (xy)/gcd(x, y)`, which is what you're doing on line 218 for `p-1` and `q-1`. should we refer to these variables as "labmda" (i.e. carmichael's totient) instead of "phi" (i.e. euler's totient) for clarity?

Reply:
This is a bad merge. Two different versions of the same check were added independently. I'll remove the new code.